### PR TITLE
Add auto-approve collection test

### LINKED ISSUE #157: ah_collection auto approve function broken
the ah_collection module has an auto_approve option but currently if set to true it will fail saying something like could not find a collection waiting for approval. I think this has to do with changes to the API/backend that needs to be updated

### PR DESCRIPTION
**Pull request by Tompage1994**
**MERGE #177 FIRST**

# What does this PR do?

Tests auto-approve collection in CI

# How should this be tested?

CI

# Is there a relevant Issue open for this?
This is looking to resolve #157 (or at least find the issue)

resolves #157

### CODE REVIEW

**Comment by sean-m-sullivan** [comment]
@Tompage1994 I am working on this in parrellel, I am looking at the approve part right now, but check
https://github.com/redhat-cop/ah_configuration/compare/devel...sean-m-sullivan:ah_configuration:toms-157-autoapprove?expand=1
, if you start work again

File: .github/workflows/ci_standalone_versioned.yml [hunk 1]
```diff
@@ -85,20 +85,21 @@ jobs:
         run: sleep 240
 
       - name: Build and install the collection
+        id: build
         uses: redhat-cop/ansible_collections_tooling/actions/build_ansible_collection@main
         with:
-          collection_namespace: infra
+          collection_namespace: redhat_cop
```
Comment (djdanielsson) [inline .github/workflows/ci_standalone_versioned.yml:91]: shouldn't this be infra?

Comment (sean-m-sullivan) [reply .github/workflows/ci_standalone_versioned.yml:91]: To make the test work on Bundeled hub and Galaxy NG, I went with redhat_cop

File: roles/collection/tasks/main.yml [hunk 2]
```diff
@@ -19,26 +21,30 @@
...
-    loop_var: __collections_job_async_result_item
-  when: __collections_job_async_result_item.ansible_job_id is defined
-  no_log: "{{ ah_configuration_user_secure_logging }}"
-  vars:
-    ansible_async_dir: '/tmp/.ansible_async'
+# - name: Sleep for 10 seconds and continue with play
+#   ansible.builtin.wait_for:
+#     timeout: 10
+
+# - name: "Update/Destroy collection | Wait for finish the update/destruction of collection"
```
Comment (sean-m-sullivan) [inline roles/collection/tasks/main.yml]: We can remove the commenting out, I was doing testing to see if it made a difference and hadn't cleaned it up yet.

Comment (sean-m-sullivan) [reply roles/collection/tasks/main.yml]: AKA put the Async stuff back in.

**Comment by Tompage1994** [comment]
I guess the problem now is what to do with V4.4 and V4.5? It seems it just doesn't work in the CI for whatever reason, so do we disable it? How do we prove it works in reality though?

**Comment by djdanielsson** [comment]
does it work in 4.4 and 4.5? a lot of the new changes I thought only worked in the latest version

**Comment by Tompage1994** [comment]
I don't think the APIs for approval have changed and I don't think we've made any updates to that anyway so it should work

**Review by sean-m-sullivan — APPROVED** [review]
LGTM I am good to merge this if you are Tom.

**Comment by Tompage1994** [comment]
Not sure I like accepting PRs when CI is broken. Do we think the best thing to do here is to disable the auto-approve check when stable-4.4 or stable-4.5?

**Comment by Tompage1994** [comment]
Ok, I've done that and disabled the auto-approve for the failing ones.

```
auto_approve: "{{ galaxy_ng_version != 'stable-4.4' and galaxy_ng_version != 'stable-4.5' }}"
```

**Review by djdanielsson — APPROVED** [review]
LGTM, is it release time?

**Pull request merged**